src/read.c

```diff
@@ -3992,50 +3992,26 @@ static avifResult avifDecoderDecodeTiles(avifDecoder * decoder, uint32_t nextIma
             if (!avifDecoderDataCopyTileToImage(decoder->data, info, decoder->image, tile, tileIndex)) {
                 return AVIF_RESULT_INVALID_IMAGE_GRID;
             }
-        }
-    }
-    return AVIF_RESULT_OK;
-}
+        } else {
+            // Non-grid path. Just steal the planes from the only "tile".
+            assert(info->tileCount == 1);
+            avifImage * src = decoder->data->tiles.tile[info->firstTileIndex].image;
+            if ((decoder->image->width != src->width) || (decoder->image->height != src->height) ||
+                (decoder->image->depth != src->depth)) {
+                if (tile->input->alpha) {
```

Comment:
most of these functions don't need to take in an `alpha` parameter since it is almost always inferable.

also, in this case we are passing separate `info` parameters for color and the alpha cases, so adding another `alpha` parameter seems explicitly redundant. so i'll leave this as-is.

Reply:
Your decision is fine. I just wanted to clarify why I asked about this.

It's because the function we deleted had an `alpha` parameter:

```
avifResult avifDecoderOutputDecodedTiles(avifDecoder * decoder, const avifTileInfo * info, avifBool alpha, unsigned int oldDecodedTileCount)
```

The `info` parameter doesn't contain an alpha boolean, so an `alpha` parameter would not be redundant with the `info` parameter.

Reply:
Got it.

I just meant that since it can be inferred from the tiles, passing an additional parameter is always going to be a bit redundant.